src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlElement.cs

```diff
@@ -68,11 +68,11 @@ public void Write(StringBuilder strBuilder, DocPosition docPos, AncestralNamespa
                 strBuilder.Append('<').Append(Name);
                 foreach (object attr in nsListToRender.GetKeyList())
                 {
-                    (attr as CanonicalXmlAttribute).Write(strBuilder, docPos, anc);
+                    (attr as CanonicalXmlAttribute)!.Write(strBuilder, docPos, anc);
                 }
                 foreach (object attr in attrListToRender.GetKeyList())
                 {
-                    (attr as CanonicalXmlAttribute).Write(strBuilder, docPos, anc);
+                    (attr as CanonicalXmlAttribute)!.Write(strBuilder, docPos, anc);
```

Comment:
I think it'd be better to fail with a cast exception rather than with a null reference exception in the event that attr is something other than the specified type.  So instead of using `!`, the foreach can be changed to be e.g.
```C#
foreach (CanonicalXmlAttribute attr in nsListToRender.GetKeyList())
```
and then just use `attr` in the loop, rather than the `as` cast and suppression.  Same for other cases above and below.

Reply:
Yes, I think so to, however I didn't change it because I thought it would change the external behaviour, which isn't encouraged in these types of PR (well, in general too, but _especially_ in these types of PRs).

Reply:
@stephentoub - you may have missed the above comment; were my assumptions incorrect?

Reply:
Thanks.  We can leave it for now and look at cleaning it up subsequently.

Reply:
OK. Should I mark this as Resolved? Or should an Issue be created to track it?